src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -49,7 +49,7 @@
 
   public boolean isConnected()
   {
-    return key.isValid() && ((SocketChannel)key.channel()).isConnected();
+    return key != null && key.isValid() && ((SocketChannel)key.channel()).isConnected();
```

Comment:
The last merged PR does not introduce the problem. The problem existed prior to the PR. The last PR relies on the fix in this PR.